Automattic/wp-calypso
# Command Palette: Unify commands

Fixes https://github.com/Automattic/wp-calypso/issues/87950

## Proposed Changes

Refactors the command palette logic so commands are no longer duplicated for Calypso and WP Admin:
- Commands are now defined in `packages/command-palette/src/commands.tsx`
- Calypso and WP Admin can override the commands when they need a specific behavior
  - Calypso: `client/sites-dashboard/components/wpcom-smp-commands.tsx`
  - WP Admin: `apps/command-palette-wp-admin/src/use-commands.ts`
- The React contexts of the Command Palette component have been merged into one single context to simplify the maintenance
- Add some tests to prevent the definition of the commands to be accidentally changed (if for instance we need to do again a refactor like this one).

## Testing Instructions

- Apply these changes to your sandbox
- Open the Calypso live link below
- Make sure there are no regressions in the command palette in both Calypso and WP Admin:
  - On a site-specific page, the current site should be automatically used without displaying the site selector
    - Check that commands are filtered accordingly (e.g. SFTP/SSH commands only show up on Atomic sites)
  - On siteless pages (e.g. `/sites` or `/read`), elegible commands should display the site selector
    - Check that sites are filtered accordingly (e.g. SFTP/SSH commands only list Atomic sites)
  - Tracks events should be registered correctly
  - Commands that check the preferred admin interface (e.g. Open general settings) should redirect to the appropriate page
  - Commands that have different behavior across Calypso and WP Admin (e.g. Get help) should maintain such behavior
